tests/testprogs/watchpoint.c

```diff
@@ -17,9 +17,19 @@ int main() {
     return 1;
   }
 
+  FILE* addr_fp = fopen("/tmp/watchpoint_mem", "w");
+
+  if (!addr_fp)
+    perror("failed to open file in /tmp");
+
+  fprintf(addr_fp, "%p", addr);
+  fclose(addr_fp);
```

Comment:
How about making this testprog print the address to stdout? Then run it in a `BEFORE` clause and cat it to a file, ie:

```
BEFORE ./testprogs/watchpoint > /tmp/watchpoint_mem &
```

Then in the `RUN` directive you can tell bpftrace about the testprog using `-p {{BEFORE_PID}}`.

Reply:
Above command in BEFORE get stuck and timeout happens. Other issue is, redirection of o/p must be handled in subprocess.Popen for BEFORE directive(changes needed).

If everything falls in place, still the race between bpftrace invocation in RUN and flushing the address to /tmp/watchpoint_mem exists.

Reply:
@danobi please review new changes